http_service/bugbug_http/models.py

```diff
@@ -332,3 +299,94 @@ def get_config_specific_groups(config: str) -> str:
     )
 
     return "OK"
+
+
+def schedule_tests_from_patch(base_rev: str, patch_hash: str) -> str:
+    from bugbug_http import REPO_DIR
+    from bugbug_http.app import JobInfo
+
+    job = JobInfo(schedule_tests_from_patch, base_rev, patch_hash)
+    LOGGER.info("Processing %s...", job)
+
+    # Retrieve the patch from Redis
+    patch_key = f"bugbug:patch:{patch_hash}"
+    patch_data_raw = redis.get(patch_key)
+
+    if not patch_data_raw:
+        LOGGER.error(f"Patch not found in Redis for hash {patch_hash}")
+        return "NOK"
+
+    hg_base_rev = utils.git2hg(base_rev)
+    LOGGER.info(f"Mapped git base rev {base_rev} to hg rev {hg_base_rev}")
+
+    # Pull the base revision to the local repository
+    LOGGER.info("Pulling base revision from the remote repository...")
+    repository.pull(REPO_DIR, "mozilla-central", hg_base_rev)
```

Comment:
```suggestion
    repository.pull(REPO_DIR, "autoland", hg_base_rev)
```

many people are basing their work on autoland, so some revisions might not be available yet on mozilla-central